builtin/logical/ssh/path_config_ca.go

```diff
@@ -48,16 +48,39 @@ func pathConfigCA(b *backend) *framework.Path {
 		Callbacks: map[logical.Operation]framework.OperationFunc{
 			logical.UpdateOperation: b.pathConfigCAUpdate,
 			logical.DeleteOperation: b.pathConfigCADelete,
+			logical.ReadOperation:   b.pathConfigCARead,
 		},
 
 		HelpSynopsis: `Set the SSH private key used for signing certificates.`,
 		HelpDescription: `This sets the CA information used for certificates generated by this
 by this mount. The fields must be in the standard private and public SSH format.
 
-For security reasons, the private key cannot be retrieved later.`,
+For security reasons, the private key cannot be retrieved later.
+
+Read operations will return the public key, if already stored/generated.`,
 	}
 }
 
+func (b *backend) pathConfigCARead(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	publicKeyEntry, err := caKey(req.Storage, caPublicKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read CA public key: %v", err)
+	}
+
+	if publicKeyEntry == nil {
+		return nil, fmt.Errorf("keys haven't been configured yet")
```

Comment:
This will cause Vault to spit out 500 errors. Use

return logical.ErrorResponse("....."), nil

instead. Up above since it'd be an error during the read it's probably fine to keep as-is and return a `500`.